src/PredictionFramework.cpp

```diff
@@ -8,11 +10,13 @@ PredictionFramework::PredictionFramework(
     const ModelInputParameters& modelInputParameters,
     const InputObservations& observations,
     Random::RNGInterface::Sptr rng,
+    const std::string& outDir,
     Utilities::logging_stream::Sptr log)
-     : seedlist_(modelInputParameters.seedlist),
-       dayShut_(modelInputParameters.day_shut),
-       modelStructure_(modelInputParameters.model_structure),
-       rng_(rng)
+     : modelInputParameters_(modelInputParameters),
```

Comment:
There are now several member variables of the `PredictionFramework` class which are unused and uninitialised (`seedlist_`, `dayShut_` etc.). If your new implementation no longer uses them, they should be deleted.